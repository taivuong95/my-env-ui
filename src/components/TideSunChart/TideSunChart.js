import React from "react";
import {
    ComposedChart,
    Line,
    Area,
    XAxis,
    YAxis,
    CartesianGrid,
    Tooltip,
    Legend,
} from "recharts";
import MoonSVG from "../UI/MoonSVG";
import SunSVG from "../UI/SunSVG";
const TideSunChart = () => {
    const data = [
        {
            name: "6h",
            sun: 0,
            tide: 1400,
        },
        {
            name: "8h",
            sun: 3000,
            tide: 1506,
        },
        {
            name: "10h",
            sun: 4500,
            tide: 989,
        },
        {
            name: "12h",
            sun: 5000,
            tide: 1228,
        },
        {
            name: "14h",
            sun: 4500,
            tide: 1100,
        },
        {
            name: "16h",
            sun: 3000,
            tide: 1700,
        },
        {
            name: "18h",
            sun: 0,
            tide: 1400,
        },
        {
            name: "20h",
            sun: 868,
            tide: 1506,
        },
        {
            name: "22h",
            sun: 868,
            tide: 989,
        },
        {
            name: "0h",
            sun: 868,
            tide: 1228,
        },
        {
            name: "2h",
            sun: 868,
            tide: 1100,
        },
        {
            name: "4h",
            sun: 868,
            tide: 1700,
        },
        {
            name: "6h",
            sun: 868,
            tide: 1700,
        },
    ];
    const CustomizedDot = (props) => {
        const { cx, cy, value } = props;
        if (value > 1350 || value === 0) {
            return <SunSVG cx={cx} cy={cy} />;
        }
        return <MoonSVG cx={cx} cy={cy} />
    };
    return (
        <ComposedChart
            width={1200} height={350}
            data={data}
            margin={{
                top: 20,
                right: 20,
                bottom: 20,
                left: 20,
            }}
        >
            <CartesianGrid stroke='#f5f5f5' />
            <XAxis dataKey='name' />
            <YAxis />
            <Tooltip />
            <Legend />
            <Area type='monotone' dataKey='tide' stroke='#8884d8' />
            <Line
                type='monotone'
                dataKey='sun'
                stroke='#ff7300'
                dot={<CustomizedDot />}
            />
        </ComposedChart>
    );
};

export default TideSunChart;
