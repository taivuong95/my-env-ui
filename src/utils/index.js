export const KelvinToCelsius = (kelvin) => {
    return kelvin - 273.15;
}