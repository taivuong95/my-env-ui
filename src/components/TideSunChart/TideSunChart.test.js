import React from 'react';
import TideSunChart from './TideSunChart';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { findByTestAttr } from '../../utils/index';

const setup = () => shallow(<TideSunChart />);
describe('TideSunChart Component', () => {
    test("expect to render Navigation component", () => {
        const wrapper = setup();
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('renders without error', () => {
        const wrapper = setup();
        const chartComponent = findByTestAttr(wrapper, "chart");
        expect(chartComponent.length).toBe(1);
    })
})