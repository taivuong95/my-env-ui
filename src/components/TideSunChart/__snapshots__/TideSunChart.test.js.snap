// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TideSunChart Component expect to render Navigation component 1`] = `
<Fragment>
  <div
    className="chart-title"
  >
     
    <span
      class="blue-title"
    >
      Tide
    </span>
    <span
      class="orange-title"
    >
      Sunrise & Sunset
    </span>
  </div>
  <ResponsiveContainer
    data-test="chart"
    debounce={0}
    height={350}
    width={1200}
  >
    <ComposedChart
      barCategoryGap="10%"
      barGap={4}
      data={
        Array [
          Object {
            "name": "6h",
            "sun": 0,
            "tide": 1400,
          },
          Object {
            "name": "8h",
            "sun": 3000,
            "tide": 1506,
          },
          Object {
            "name": "10h",
            "sun": 4500,
            "tide": 989,
          },
          Object {
            "name": "12h",
            "sun": 5000,
            "tide": 1228,
          },
          Object {
            "name": "14h",
            "sun": 4500,
            "tide": 1100,
          },
          Object {
            "name": "16h",
            "sun": 3000,
            "tide": 1700,
          },
          Object {
            "name": "18h",
            "sun": 0,
            "tide": 1400,
          },
          Object {
            "name": "20h",
            "sun": 868,
            "tide": 1506,
          },
          Object {
            "name": "22h",
            "sun": 868,
            "tide": 989,
          },
          Object {
            "name": "0h",
            "sun": 868,
            "tide": 1228,
          },
          Object {
            "name": "2h",
            "sun": 868,
            "tide": 1100,
          },
          Object {
            "name": "4h",
            "sun": 868,
            "tide": 1700,
          },
          Object {
            "name": "6h",
            "sun": 868,
            "tide": 1700,
          },
        ]
      }
      layout="horizontal"
      margin={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      reverseStackOrder={false}
      stackOffset="none"
      syncMethod="index"
    >
      <CartesianGrid
        fill="none"
        horizontal={true}
        horizontalFill={Array []}
        horizontalPoints={Array []}
        stroke="#f5f5f5"
        vertical={true}
        verticalFill={Array []}
        verticalPoints={Array []}
      />
      <XAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        dataKey="name"
        domain={
          Array [
            0,
            "auto",
          ]
        }
        height={30}
        hide={false}
        mirror={false}
        orientation="bottom"
        padding={
          Object {
            "left": 0,
            "right": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="category"
        width={0}
        xAxisId={0}
      />
      <YAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        domain={
          Array [
            0,
            "auto",
          ]
        }
        height={0}
        hide={false}
        mirror={false}
        orientation="left"
        padding={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="number"
        width={60}
        yAxisId={0}
      />
      <Tooltip
        active={false}
        allowEscapeViewBox={
          Object {
            "x": false,
            "y": false,
          }
        }
        animationDuration={400}
        animationEasing="ease"
        contentStyle={Object {}}
        coordinate={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cursor={true}
        cursorStyle={Object {}}
        filterNull={true}
        isAnimationActive={true}
        itemStyle={Object {}}
        labelStyle={Object {}}
        offset={10}
        separator=" : "
        trigger="hover"
        useTranslate3d={false}
        viewBox={
          Object {
            "x1": 0,
            "x2": 0,
            "y1": 0,
            "y2": 0,
          }
        }
        wrapperStyle={Object {}}
      />
      <Legend
        align="center"
        iconSize={14}
        layout="horizontal"
        verticalAlign="bottom"
      />
      <Area
        activeDot={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={false}
        dataKey="tide"
        dot={false}
        fill="#3182bd"
        fillOpacity={0.6}
        hide={false}
        isAnimationActive={true}
        legendType="line"
        points={Array []}
        stroke="#8884d8"
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Line
        activeDot={true}
        animateNewValues={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={false}
        dataKey="sun"
        dot={<CustomizedDot />}
        fill="#fff"
        hide={false}
        isAnimationActive={true}
        legendType="line"
        points={Array []}
        stroke="#ff7300"
        strokeWidth={1}
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
    </ComposedChart>
  </ResponsiveContainer>
</Fragment>
`;
